### PR TITLE
Iar 36/tomas/create a new subreddit fe unit tests

### PR DESCRIPTION
**Pull request by TomasBenes**
Upgraded functionality for creating subreddit plus unit tests.

### CODE REVIEW

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.spec.ts [hunk 1]
```diff
@@ -1,31 +1,55 @@
...
 import { AddNewSubredditComponent } from './add-new-subreddit.component';
 import { GatewayService } from 'src/app/services/gateway.service';
+import { SubredditCreation } from 'src/types/subreddits';
+import { ModuleWithProviders } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { RouterTestingModule } from '@angular/router/testing';
+import { FeedComponent } from '../feed/feed.component';
 
-class MockDataService {}
+class MockService {
```
Comment (SimProch) [inline src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.spec.ts:13]: MockService doesnt say much about what is being mocked.

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts [hunk 2]
```diff
@@ -15,17 +17,16 @@ export class AddNewSubredditComponent {
         description: '',
     };
 
-    constructor(private _gatewayService: GatewayService) {}
+    errorMessage?: Error;
```
Comment (SimProch) [inline src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts:20]: errorMessage sounds like it is a string rather than an object.

**Review by SimProch — CHANGES_REQUESTED** [review]

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.spec.ts (hunk 1 above)
Comment (TomasBenes) [reply src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.spec.ts]: changed name as requested

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts (hunk 2 above)
Comment (TomasBenes) [reply src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts]: changed name as requested

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts [hunk 3]
```diff
@@ -15,17 +17,16 @@ export class AddNewSubredditComponent {
         description: '',
     };
 
-    constructor(private _gatewayService: GatewayService) {}
+    errorObject?: Error;
```
Comment (SimProch) [inline src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts:20]: errorObject does not need the "Object" part as it is redundant. Making this "error" is sufficient. Ill approve the PR, but update it.

**Review by SimProch — APPROVED** [review]

File: src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts (hunk 3 above)
Comment (TomasBenes) [reply src/app/components/authorized/add-new-subreddit/add-new-subreddit.component.ts]: changed name as requested

**Review by Karhalll — APPROVED** [review]
Looks very ok for me :)

**Pull request merged**